### PR TITLE
UI Simplification: Allow applying Presets

### LINKED ISSUE #10609: UI Simplification: Text element insertion panel - Allow applying Presets
## Feature Description
In addition to adding an element by clicking on a text Preset, it should also be possible to apply it, if a text element is selected.

Note: this feature should be released together with #10113 and not separately!

- [x] Clicking on a Preset when no element is selected or non-text element is selected: Insert a new text element
- [x] Clicking on a Preset when text element selected: Apply the preset to the selected text(s)
- [x] Note: Do not display the Insertion `+` icon on the Preset if clicking on it results in applying style. The insertion icon should be displayed only if we'll add a new element when clicking.
- [ ] Performance: make sure only the Presets re-render when selecting/de-selecting a text element, and make sure it re-renders only if it actually needs to.
- [x] Update tests (quite a few will break due to not being able to insert two elements in a row anymore)
- [x] Remove feature flag for Saved Styles added in #10113 

Note: for consecutively adding text elements, the user now has to de-select text first and then click to add text again.

### PR DESCRIPTION
**Pull request by miina**
## Summary
Allows applying presets when text element is selected (or multiple text elements).

## To-do

- [x] Remove the Saved Styles feature flag once #10685 gets merged.
- [x] Update tests
- [x] Add new tests

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Add a text element, for example Paragraph. Verify that `+` icon is displayed when hovering over the Paragraph preset.
2. Now keep the text element selected and hover over Title 1. Verify the `+` icon is not displayed anymore.
3. Click on the preset. Verify that instead of adding a new one, the preset is applied to the existing element instead -- e.g. the font size changes, etc.
4. Try the same with other presets / multiple selected texts -- when text is selected, the Preset is applied; when something else is selected (or nothing), a new element is inserted.

### Does this PR have a security-related impact?
No

### Does this PR change what data or activity we track or use?
No

### Does this PR have a legal-related impact?
No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10609

### CODE REVIEW

File: packages/story-editor/src/components/canvas/karma/quickActions.karma.js
Comment (miina) [inline packages/story-editor/src/components/canvas/karma/quickActions.karma.js]: Note: skipping the tests is temporary in this PR, will add these back in once `main` passes -- it was necessary for detecting which tests failed due to this PR and which not.

File: packages/story-editor/src/components/library/panes/text/fontPreview.js
Comment (barklund) [inline packages/story-editor/src/components/library/panes/text/fontPreview.js]: Nit: This technically needs a Boolean wrapper to be consistently a bool.

```js
    const isText =
      Boolean(state.selectedElements && areAllType('text', state.selectedElements));
```

File: packages/story-editor/src/components/library/panes/text/fontPreview.js
Comment (barklund) [inline packages/story-editor/src/components/library/panes/text/fontPreview.js:198]: Why is the element being moved as well when applying a text preset? Should the element stay in place?

preset-move

And should things like italic, color, underline, all caps, and text alignment also be "reset" rather than just font family, font size, and font weight?

File: packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js
Comment (barklund) [inline packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: Can there be more than one?

Comment (barklund) [reply packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: Ah, it is from the two different panels - because the text exists in both the text insertion panel and the design panel?

Perhaps it should have a test for each separately, to confirm that it exists in both panels individually. That will also make the test (correctly) fail once the panels are merged in #10111.

**Review by barklund — COMMENTED** [review]
Seems to work fine, though I'm not convinced that all the right properties of the saved elements are applied to the existing texts.

File: packages/story-editor/src/components/library/panes/text/fontPreview.js
Comment (miina) [reply packages/story-editor/src/components/library/panes/text/fontPreview.js:198]: No, the element should definitely not move around!

> And should things like italic, color, underline, all caps, and text alignment also be "reset" rather than just font family, font size, and font weight

Not certain about it -- will check with UX.

File: includes/Experiments.php
Comment (swissspidy) [inline includes/Experiments.php:392]: We usually first enable a feature by default before removing the flag code completely. But I suppose for this small change here it's fine.

Comment (miina) [reply includes/Experiments.php:392]: I know, I wasn't planning to add it behind a feature flag originally at all since it's quite a small change but since UX preferred having it released together with this feature (Applying presets), then added it anyway for ensuring these two get released together.

File: packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js
Comment (miina) [reply packages/story-editor/src/components/panels/design/textStyle/karma/stylePresets.karma.js]: This test should actually really test for the one in the Library. Adjusted the test to do that.

File: packages/story-editor/src/components/library/panes/text/fontPreview.js
Comment (miina) [reply packages/story-editor/src/components/library/panes/text/fontPreview.js:198]: For visibility: looks like at least the colors, font family, and text-alignment should be kept as it was in the original element.
Waiting for confirmation from @agingoldseco, see the [Slack discussion](https://xwp.slack.com/archives/C01ESBB8ZN1/p1646210559348009).

EDIT: As of current information, only the font color should be kept from the inline styles.

**Comment by merapi** [comment]
It might be connected to the existing issue (shouldn't set the padding to Mixed when the color is changed, right?):

https://user-images.githubusercontent.com/21036224/156668910-a5b0ad01-2898-4bf3-abd5-c4fddfac201f.mov

**Comment by miina** [comment]
> It might be connected to the existing issue (shouldn't set the padding to Mixed when the color is changed, right?):

Seems like a separate issue -- I've seen odd padding values as well (`null`) but have seen it quite randomly without understanding where it came exactly. I'll create a follow-up issue for this as it doesn't seem to be related to Applying presets.

**Comment by miina** [comment]
@barklund @merapi This PR is ready for final review now -- got confirmation from UX that the text color should be kept but most of the other Text Styles not. The others would be overwritten in a similar way as the Saved Styles overwrite most of the props (but not 100% all) (yep, it could potentially cause confusion).

File: packages/story-editor/src/components/library/panes/text/fontPreview.js
Comment (merapi) [reply packages/story-editor/src/components/library/panes/text/fontPreview.js]: Maybe we can just use `isText: areAllType('text', state.selectedElements),` because in the worst case `selectedElements = STABLE_ARRAY` and `Boolean(STABLE_ARRAY || any array) === true`.

**Comment by merapi** [comment]
When using the keyboard there is no visible focus when you are in the "apply preset" mode.

I guess it should look like the focus on the Saved Styles:

image

**Comment by miina** [comment]
> When using the keyboard there is no visible focus when you are in the "apply preset" mode.

Good catch! Fixed.

File: packages/story-editor/src/components/library/panes/text/fontPreview.js
Comment (miina) [reply packages/story-editor/src/components/library/panes/text/fontPreview.js]: This extra `state.selectedElements &&` is for checking if the `state.selectedElements` is set at all, if it's `null` then `areAllType` would error since it's checking for `selectedElements.length` inside. Thoughts?

Comment (merapi) [reply packages/story-editor/src/components/library/panes/text/fontPreview.js]: My point is: is it possible that `selectedElements` can be `null`?
https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/story-editor/src/app/story/storyProvider.js#L96

Comment (miina) [reply packages/story-editor/src/components/library/panes/text/fontPreview.js]: Perhaps not! For some reason I remember it erroring without checking first but doesn't seem to happen currently. Changing.

**Review by merapi — APPROVED** [review]
This will be very confusing for long term users but the PR itself is 🏅

**Pull request merged**
